### PR TITLE
Adding advanced usage docs

### PR DESCRIPTION
**Pull request by devin-petersohn**
## Related issue number

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] passes `black --check modin/`

### CODE REVIEW

File: docs/using_modin.rst [hunk 1]
```diff
@@ -62,6 +62,64 @@ Using Modin on a Cluster
 Modin can be run on a cluster, but the setup process is quite complex. We are working on
 a solution to make Modin run on a cluster with a simple setup. More on this coming soon!
 
+Advanced usage (experimental)
+-----------------------------
+
+In some cases, it may be useful to customize your Ray environment. Below, we have listed
+a few ways you can solve common problems in data management with Modin by customizing
+your Ray environment. It is possible to use any of Ray's initialization parameters, all
```
Comment (williamma12) [inline docs/using_modin.rst]: should be "which are all found in Ray's documentation"

File: docs/using_modin.rst [hunk 2]
```diff
@@ -62,6 +62,64 @@ Using Modin on a Cluster
...
+   import modin.pandas as pd
+
+Modin will automatically connect to the Ray instance that is already running. This way,
+you can customize your Ray environment for use in Modin!
+
+Exceeding memory (Out of core pandas)
+"""""""""""""""""""""""""""""""""""""
+
+If you are working with very large files or would like to exceed your memory, you may
+change the primary location of the DataFrame. If you would like to exceed memory, you
```
Comment (williamma12) [inline docs/using_modin.rst]:
```suggestion
change the primary location of the DataFrame or you
```

File: docs/using_modin.rst [hunk 3]
```diff
@@ -62,6 +62,64 @@ Using Modin on a Cluster
...
+
+Here is how you can exceed memory by using your disk:
+
+.. code-block:: python
+
+   import ray
+   num_bytes = 2**40 # Make sure you have disk space to do this!
+   ray.init(plasma_directory="/tmp", object_store_memory=num_bytes)
+   import modin.pandas as pd
+
```
Comment (osalpekar) [inline docs/using_modin.rst]: Suggestion for next line:
Setting ``plasma_directory="/tmp"`` uses your disk for storing the DataFrame and setting `object_store_memory` sets the maximum size of the plasma store.

File: docs/using_modin.rst [hunk 4]
```diff
@@ -62,6 +62,64 @@ Using Modin on a Cluster
...
+the number of bytes will set the maximum size of the plasma store.
+
+**Warning: Make sure you have enough space in your disk for however many bytes you**
+**request for your DataFrame**
+
+Reducing or limiting the resources Modin can use
+""""""""""""""""""""""""""""""""""""""""""""""""
+
+By default, Modin will use all of the resources available on your machine. It is
+possible, however, to limit the amount of resources Modin uses if you wish to free up
```
Comment (williamma12) [inline docs/using_modin.rst]:
```suggestion
possible, however, to limit the amount of resources Modin uses to free up
```

**Pull request merged**